### ISSUE TITLE
Include columnType in GridFilterItem

### ISSUE THREAD

**Issue by levireciprocity** [mui/mui-x#11809]
### Search keywords

columType GridFilterItem GridFilterModel server-side filtering

### Latest version

- [X] I have tested the latest version

### Summary

Add the columnType field to the GridFilterItem interface on the GridFilterModel, this field would help for server-side datagrid filtering

### Motivation

make server-side datagrid filtering more generic

**Comment by michelengelen** [comment]
What exactly are you trying to do?
Could you give an example in code?

Just asking because you can get the type from your column definition with the mandatory `field` property on the [`GridFilterItem`](https://next.mui.com/x/api/data-grid/grid-filter-item/).

It could look something like this:

```tsx
const handleFilterModelChange = (model: GridFilterModel) => {
  const modelItemsWithType: Array<GridFilterItem & { type: GridColDef['type'] }> =
    model.items.map((item) => ({
      ...item,
      type: apiRef.current.getColumn(item.field).type,
    }));
  console.log('itemsWithType', modelItemsWithType);
};
```

Here is a working example of that handler: [FilterModelItems with `type`](https://stackblitz.com/edit/react-e8injy?file=Demo.tsx)

**Comment by levireciprocity** [comment]
Yeah this is the solution I have, rather than mapping through the items to get the type I'd like to just have the type on the GridFilterItem with operatorValue, value, columnField and id

**Comment by michelengelen** [comment]
OK, imo this should be sufficient and not blur the API even more, but I'll put this on the board nonetheless for the team to look at.
